composite_file/Composite/src/Client.java

```diff
@@ -0,0 +1,23 @@
+public class Client {
+    public static void main(String[] args) {
+        INameProvider file = new File("|-----> Adapter");
```

Comment:
why do you need to define this `|----->` ? this should be accomplished by the structure that you are building with the file and directory mixture